plugins/woocommerce/src/Caches/OrderCache.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Automattic\WooCommerce\Caches;
+
+use Automattic\WooCommerce\Caching\ObjectCache;
+
+/**
+ * A class to cache order objects.
+ */
+class OrderCache extends ObjectCache {
+
+	/**
+	 * Get the identifier for the type of the cached objects.
+	 *
+	 * @return string
+	 */
+	public function get_object_type(): string {
+		return 'orders';
+	}
+
+	/**
+	 * Get the id of an object to be cached.
+	 *
+	 * @param array|object $object The object to be cached.
+	 * @return int|string|null The id of the object, or null if it can't be determined.
+	 */
+	protected function get_object_id( $object ) {
+		return $object->get_id();
+	}
+
+	/**
+	 * Validate an object before caching it.
+	 *
+	 * @param array|object $object The object to validate.
+	 * @return string[]|null An array of error messages, or null if the object is valid.
+	 */
+	protected function validate( $object ): ?array {
+		if ( ! $object instanceof \WC_Abstract_Legacy_Order ) {
+			return array( 'The supplied order is not an instance of WC_Order, ' . gettype( $object ) );
```

Comment:
```suggestion
			return array( 'The supplied order is not an instance of WC_Abstract_Legacy_Order, ' . gettype( $object ) );
```

Maybe more accurate?

Reply:
Addressed in https://github.com/woocommerce/woocommerce/pull/35014/commits/b5ae765b9311a88d94b24154358fbdde59e872d6